crates/dekaf/Cargo.toml

```diff
@@ -15,25 +15,49 @@ doc = { path = "../doc" }
 gazette = { path = "../gazette" }
 labels = { path = "../labels" }
 ops = { path = "../ops" }
+flowctl = { path = "../flowctl" }
```

Comment:
Out of curiosity, where do you think the referred code should live then? Specifically it's [`flowctl::pagination`](https://github.com/estuary/flow/blob/master/crates/flowctl/src/pagination.rs) which is used to read items from postgrest using the pagination API. We don't seem to have a `utils` crate or anything like that 🤔

Reply:
A little duplication is better than a little dependency.

Or, live with the smell for now, but with an active plan to improve it later (we're intending to fetch singleton materialization specs after all; or, perhaps you have an opinion on a better factoring, or perhaps a better spot will become apparent later, or maybe the postgrest crate has better utilities for pagination now?).

Reply:
Just to fine-tune my "inner Johnny" a bit,  what's the problem with this dependency? Is it about build times, i.e pulling in all of flowctl (and its dependencies) just for one module? Or is it some more fundamental argument?

Reply:
I'm a little baffled here. Presume a goal of a well-structured code base with an intuitive dependency graph. Turning the question around, finish the sentence "It's reasonable for `dekaf` to depend on our end-user CLI command because ___".

Reply:
I think I'd like to leave it for now, just to avoid creeping the scope of this PR even more. It's a pretty tight dependency on a single isolated module within `flowctl`. Prior to this PR, `flowctl` is the only crate that directly called out to PostgREST, so it made sense to include utilities such as pagination directly in it. In terms of well-structured code with an intuitive dependency graph, now that we have two "leaf" crates that need to share this functionality, it does make more sense to move it to somewhere shared by both.